Catappult/appcoins-wallet-android
# [feature/ESKL-140] - Add queue ID support for wallet

**What does this PR do?**

Allow users to create a "room" so that they can play against their friends. This meant a refactor on the payments side, which means that we are no longer using OSP URIs to process the payment. Instead, e-Skills module is now in charge of dealing also with the payment. Nevertheless, the actual implementation is hidden on the app module.

This PR includes a bump on targetSdkVersion and on room_version that I needed in order to compile on the M1 mac. You can revert this as it has no impact on the PR itself.

**Database changed?**

   No

**Where should the reviewer start?**


**How should this be manually tested?**

With 2048 dev version, select "Play for rewards" and try to purchase. Try the following flows:
- Select "BUY" right away
- Insert a room name and check if it appears on the loading screen
- Try to copy/paste room name

**What are the relevant tickets?**

  Tickets related to this pull-request: [ESKL-140](https://aptoide.atlassian.net/browse/ESKL-140)

**Code Review Checklist**
